sdk/core/azure-core/src/test/java/com/azure/android/core/util/DateTimeRfc1123Test.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.util;
+
+import org.junit.Test;
+import org.threeten.bp.LocalDateTime;
+import org.threeten.bp.OffsetDateTime;
+import org.threeten.bp.ZoneId;
+import org.threeten.bp.ZoneOffset;
+import org.threeten.bp.format.DateTimeFormatter;
+
+import java.util.Locale;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class DateTimeRfc1123Test {
+    private static final DateTimeFormatter RFC1123_DATE_TIME_FORMATTER =
+        DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss 'GMT'").withZone(ZoneId.of("UTC")).withLocale(Locale.US);
+    private static final String TEST_DATE = "Tue, 25 Feb 2020 00:59:22 GMT";
+
+    @Test
+    public void constructor_withOffsetDateTime() {
+        DateTimeRfc1123 dateTimeRfc1123 = new DateTimeRfc1123(
+            OffsetDateTime.of(LocalDateTime.parse(TEST_DATE, RFC1123_DATE_TIME_FORMATTER), ZoneOffset.UTC));
+
+        assertEquals(TEST_DATE, dateTimeRfc1123.getDateTime().format(RFC1123_DATE_TIME_FORMATTER));
+    }
+
+    @Test
+    public void constructor_withString() {
+        DateTimeRfc1123 dateTimeRfc1123 = new DateTimeRfc1123(TEST_DATE);
+
+        assertEquals(TEST_DATE, dateTimeRfc1123.getDateTime().format(RFC1123_DATE_TIME_FORMATTER));
+    }
+
+    @Test
+    public void convertToString() {
+        DateTimeRfc1123 dateTimeRfc1123 = new DateTimeRfc1123(
+            OffsetDateTime.of(LocalDateTime.parse(TEST_DATE, RFC1123_DATE_TIME_FORMATTER), ZoneOffset.UTC));
+
+        assertEquals(TEST_DATE, dateTimeRfc1123.toString());
+    }
+
+    @Test
+    public void getHashCode() {
+        DateTimeRfc1123 dateTimeRfc1123 = new DateTimeRfc1123(
+            OffsetDateTime.of(LocalDateTime.parse(TEST_DATE, RFC1123_DATE_TIME_FORMATTER), ZoneOffset.UTC));
+
+        assertEquals(1468106660, dateTimeRfc1123.hashCode());
```

Comment:
Unless we have overridden the hash code method I wouldn't eagerly test this as it is dependent on the underlying JVM and has the potential for collisions.

Reply:
We did override it in our `DateTimeRfc1123` class, however, it just uses the `hashCode` method in `OffsetDateTime`. I think I will remove this test for the time being.

Reply:
Actually, after digging around on the internet, I think I will take the approach showed [here](https://stackoverflow.com/questions/4449728/how-can-i-do-unit-test-for-hashcode).